lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -96,31 +99,45 @@ def validate_secret_spec(secret_name, spec)
     end
 
     def create_or_update_secret(secret_name, secret_type, data)
-      metadata = {
-        name: secret_name,
-        labels: { "name" => secret_name },
-        namespace: @namespace,
-        annotations: { MANAGEMENT_ANNOTATION => "true" }
-      }
-      secret = Kubeclient::Secret.new(type: secret_type, stringData: data, metadata: metadata)
-      if secret_exists?(secret)
-        KubernetesDeploy.logger.info("Updating secret #{secret_name}")
-        @kubeclient.update_secret(secret)
-      else
-        KubernetesDeploy.logger.info("Creating secret #{secret_name}")
-        @kubeclient.create_secret(secret)
+      secret_yaml = generate_secret_yaml(secret_name, secret_type, data)
+      msg = secret_exists?(secret_name) ? "Updating secret #{secret_name}" : "Creating secret #{secret_name}"
+      KubernetesDeploy.logger.info(msg)
+
+      file = Tempfile.new(secret_name)
+      file.write(secret_yaml)
+      file.close
```

Comment:
To address both this and @mkobetic comment below, kubectl won't be able to read the file unless we close it. Tempfiles are automatically GC'd when the process exits, if not sooner. `Tempfile.open { |f| ... }` doesn't return the file. I don't think it would be possible for the file to get GC'd when we need it in this case, since we only need it for the duration of this method, which retrains a reference to it. However, I can switch to a regular file and manually unlink it in an `ensure` to be extra super safe.